### DISCUSSION TITLE
Add Angle2D and Angle3D classes to represent angles and rotation.

### DISCUSSION THREAD

**Discussion by ghost** [godotengine/godot-proposals#8837] [Engine Core]
The way Godot handles angles and rotation right now, with simple floats (like `rotation` and `rotation_degrees`), vectors and functions (like `rad_to_deg()`), is pretty rudimentary considering the otherwise super-high level of the engine. Making standardized `Angle2D` and `Angle3D` classes that have functions for working with different angle types (degree, radians, normalized...), while storing the values in a single matter, seems like an optimization that could enhance workflow and reduce the number of errors.

**Comment by AThousandShips** [comment]
See:
* https://github.com/godotengine/godot-proposals/issues/5397

Which had more details but was closed due to lack of support

Also other than rotation there's no simple definition of "angle" in three dimensions, only concepts of rotation in specific coordinate systems

**Reply by ghost** [reply]
Oh, I looked everywhere for a similar suggestion. Thank you for pointing it out!

**Reply by AThousandShips** [reply]
Make sure to search closed ones too, they are easy to forget 🙂

**Comment by Mickeon** [comment]
Angle3D? There's so many ways to express 3D angles in Godot. **Basis**, **Quartenions**, a **Vector3** containing euler angles in either radians or degrees... I agree they _look_ really complicated on the surface, and they most certainly are, but I feel like it doesn't warrant adding yet another type for them.
One could say the introduction to 3D angles in Godot is not exactly smooth, and it's that aspect that needs to be improved upon.
